ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -609,32 +609,25 @@ static void endToEndBackupManagerTest(MockCluster cluster, DataNodeId dataNode,
         clientSSLConfig == null ? "" : dataNode.getDatacenterName());
     props.setProperty("clustermap.port", "12309");
     props.setProperty("vcr.ssl.port", "12310");
+    if (clientSSLConfig == null) {
+      props.setProperty("ssl.factory", MockSSLFactory.class.getName());
```

Comment:
I see the problem here: SSL port in `AmbryDataNode` is not required but required in `CloudDataNode`. 

Talked with @pnarayanan , though SSL is always required in Li, it's optional in open source because people may not have SSL configed in their Prod. 
Same to VCR, people may not have SSL either, so let's make SSL in VCR optional?

Reply:
This also syncs the SSL logic in AmbryServer and VCRServer.

Reply:
Made it optional.